go/interpreter/lfvm/memory_test.go

```diff
@@ -357,3 +355,143 @@ func TestToValidMemorySize_RoundsUpToNextMultipleOf32(t *testing.T) {
 		}
 	}
 }
+
+func TestMemory_set_SuccessfulCase(t *testing.T) {
+	const (
+		memoryOriginalSize = uint64(8)
+		offset             = uint64(1)
+		sizeToExpand       = 32 - memoryOriginalSize - offset
+	)
+	// data of size (memoryOriginalSize - offset), to ensure it would fit in memory
+	data := make([]byte, memoryOriginalSize-offset)
+	for i := range data {
+		// add non zero values.
+		data[i] = byte(i + 1)
+	}
+	size := uint64(len(data))
+	c := context{gas: 3}
+	m := NewMemory()
+	m.store = []byte{0xff}
+
+	err := m.set(offset, size, data, &c)
+	if err != nil {
+		t.Fatalf("unexpected error, want: %v, got: %v", nil, err)
+	}
+	if m.length() != toValidMemorySize(memoryOriginalSize) {
+		t.Errorf("set should not change memory size, want: %d, got: %d", toValidMemorySize(memoryOriginalSize), m.length())
+	}
+	want := append([]byte{0xff}, data...)
+	want = append(want, []byte{sizeToExpand: 0x0}...)
+	if !bytes.Equal(m.store, want) {
+		t.Errorf("unexpected memory value, want: %x, got: %x", want, m.store)
+	}
+}
+
+func TestMemory_set_ErrorCases(t *testing.T) {
+	c := context{gas: 0}
+	m := NewMemory()
+
+	err := m.set(0, 1, []byte{}, &c)
+	if !errors.Is(err, errOutOfGas) {
+		t.Errorf("unexpected error, want: %v, got: %v", errOutOfGas, err)
+	}
+	err = m.set(math.MaxUint64, 1, []byte{}, &c)
+	if !errors.Is(err, errGasUintOverflow) {
+		t.Errorf("unexpected error, want: %v, got: %v", errGasUintOverflow, err)
+	}
+	err = m.set(1, math.MaxUint64, []byte{}, &c)
+	if !errors.Is(err, errGasUintOverflow) {
+		t.Errorf("unexpected error, want: %v, got: %v", errGasUintOverflow, err)
+	}
+}
+
+func TestMemory_set_sizeZeroDoesNotProduceExpansion(t *testing.T) {
+	for offset := uint64(0); offset < 128; offset++ {
+		c := context{gas: 0}
+		m := NewMemory()
+		m.store = []byte{0x1}
+		err := m.set(offset, 0, []byte{}, &c)
+		if err != nil {
+			t.Errorf("unexpected error: %v", err)
+		}
+		if m.length() != 1 {
+			t.Errorf("memory should not have been expanded, instead is: %d", m.length())
+		}
+	}
+}
+
+func TestMemory_setByte_SuccessfulCase(t *testing.T) {
+	memory := []byte{0x12, 0x34, 0x56, 0x78}
+	offset := uint64(8)
+	ctxt := context{gas: 3}
+	m := NewMemory()
+	m.store = bytes.Clone(memory)
+	value := byte(0xff)
+
+	err := m.setByte(offset, value, &ctxt)
+	if err != nil {
+		t.Errorf("unexpected error, want: %v, got: %v", nil, err)
+	}
+	if m.length() <= offset {
+		t.Errorf("unexpected memory size, want: %d, got: %d", offset, m.length())
+	}
+	for i, b := range memory {
+		if i == int(offset) {
```

Comment:
`i` is never the offset ... the offset is `8` and `i` is only in the range 0-3.

Reply:
right this was supposed to range in `m.store`, will fix.